### PR TITLE
Flask API: Add implementation for CIS-CAT endpoints

### PR DESCRIPTION
**Pull request by mgmacias95**
Hello team,

This PR closes https://github.com/wazuh/wazuh/issues/2840.

This PR also implements parsing for `search` and `sort` parameters. The API receives a string but the framework expects a dictionary. The `select` parameter has been changed in the framework: it used to be a dictionary but now it's just a list.

Best regards,
Marta

### CODE REVIEW

File: api/api/util.py [hunk 1]
```diff
@@ -154,6 +154,39 @@ def remove_nones_to_dict(dct):
...
+    """Parses an str parameter from the API query and returns a dictionary the framework can process
+
+    :param param: Str parameter coming from the API.
+    :param param_type: Type of parameter -> search or sort
+    :return: A dictionary
+    """
+    if param is not None:
+        return parse_search_param(param) if param_type == 'search' else parse_sort_param(param)
+    else:
+        return None
```
Comment (crd1985) [inline api/api/util.py]: I would change this wraper into something like:
```
my_func = f'_parse_{param_type}'
parser = globals().get(my_func, default=lambda x: param)
return parser(param)
```

File: api/api/controllers/ciscat_controller.py [hunk 2]
```diff
@@ -2,6 +2,71 @@
...
+from typing import List
 
-def get_agents_cistat_results():
-    pass
+import wazuh.ciscat as ciscat
+from wazuh.cluster.dapi.dapi import DistributedAPI
+from ..util import remove_nones_to_dict, parse_api_param
+
+loop = asyncio.get_event_loop()
+logger = logging.getLogger('wazuh')
```
[the remark this answers is not in the document]
Comment (mgmacias95) [reply api/api/controllers/ciscat_controller.py]: The wazuh logger must be used. The child logger doesn't inherit all properties from the father such as logging filters, formats, or the custom debug2 level.

File: api/api/util.py (hunk 1 above)
Comment (mgmacias95) [reply api/api/util.py]: Done in ec0e3e9cff595db5b614e0050c0dd7d7288b6336

**Comment by davidjiglesias** [comment]
* Integration test results:
```
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.4.1, py-1.8.0, pluggy-0.9.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /var/ossec/framework/python/bin
plugins: tavern-0.26.3
collected 9 items                                                              

test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results PASSED [ 11%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[profile] PASSED [ 22%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[score] PASSED [ 33%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[error] PASSED [ 44%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[fail] PASSED [ 55%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[benchmark] PASSED [ 66%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[pass] PASSED [ 77%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[notchecked] PASSED [ 88%]
test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[unknown] PASSED [100%]

=============================== warnings summary ===============================
/var/ossec/framework/python/lib/python3.7/site-packages/PyYAML-3.13-py3.7-linux-x86_64.egg/yaml/constructor.py:126
  /var/ossec/framework/python/lib/python3.7/site-packages/PyYAML-3.13-py3.7-linux-x86_64.egg/yaml/constructor.py:126: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working

test/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results
  /var/ossec/framework/python/lib/python3.7/site-packages/jwt/api_jwt.py:81: DeprecationWarning: It is strongly recommended that you pass in a value for the "algorithms" argument when calling decode(). This argument will be mandatory in a future version.
    DeprecationWarning

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===================== 9 passed, 2 warnings in 0.83 seconds =====================
```

**Comment by davidjiglesias** [comment]
* Unit test results:
```
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.4.2, py-1.8.0, pluggy-0.11.0
rootdir: /opt/project/framework
plugins: tavern-0.26.3collected 1 item

test_ciscat.py .                                                         [100%]

=========================== 1 passed in 0.60 seconds ===========================
```

**Comment by davidjiglesias** [comment]
* Updated and run unit test:
```
============================ test session starts ============================
platform linux -- Python 3.7.2, pytest-4.5.0, py-1.8.0, pluggy-0.11.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /var/ossec/framework/python/bin
plugins: tavern-0.26.3, cov-2.7.1
collected 2 items                                                           

unit/test_ciscat.py::TestCiscat::test_get_ciscat_results PASSED       [ 50%]
unit/test_ciscat.py::TestCiscat::test_get_results_agent PASSED        [100%]

========================= 2 passed in 0.12 seconds ==========================
```

**Comment by davidjiglesias** [comment]
Updated test results with changes to environment

**Integration tests:**
```
============================= test session starts ==============================
platform linux -- Python 3.6.7, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/davidjiglesias/git/wazuh/api
plugins: tavern-0.26.4
collected 9 items                                                              

test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results PASSED [ 11%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[profile] PASSED [ 22%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[score] PASSED [ 33%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[error] PASSED [ 44%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[fail] PASSED [ 55%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[benchmark] PASSED [ 66%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[pass] PASSED [ 77%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[notchecked] PASSED [ 88%]
test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results[unknown] PASSED [100%]

=============================== warnings summary ===============================
test/integration/test_ciscat_endpoints.tavern.yaml::GET /ciscat/{agent_id}/results
  /home/davidjiglesias/.local/lib/python3.6/site-packages/jwt/api_jwt.py:81: DeprecationWarning: It is strongly recommended that you pass in a value for the "algorithms" argument when calling decode(). This argument will be mandatory in a future version.
    DeprecationWarning

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================== 9 passed, 1 warnings in 176.52 seconds ====================
```
**Unit tests:**
```
==================================== test session starts ====================================
platform linux -- Python 3.7.2, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /var/ossec/framework/python/bin
plugins: cov-2.7.1, tavern-0.26.4
collected 1 item                                                                            

unit/test_ciscat.py::test_get_ciscat_results PASSED                                   [100%]

================================= 1 passed in 0.11 seconds ==================================
```

**Comment by davidjiglesias** [comment]
Updated Integration tests results:

**Pull request merged**